components/button/test/floating-buttons.visual-diff.js

```diff
@@ -0,0 +1,48 @@
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-floating-buttons', function() {
+
+	const visualDiff = new VisualDiff('floating-buttons', __dirname);
+
+	let browser, page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.setViewport({width: 800, height: 800, deviceScaleFactor: 2});
+		await page.goto(`${visualDiff.getBaseUrl()}/components/button/test/floating-buttons.visual-diff.html`, {waitUntil: ['networkidle0', 'load']});
+		await page.bringToFront();
+	});
+
+	after(() => browser.close());
+
+	[
+		{category: 'normal', tests: ['normal', 'hover', 'focus', 'disabled']},
```

Comment:
Since buttons are ultimately responsible for testing their own hover/focus/disabled state, I think we can exclude all of these cases from here. I'd just worry about testing the two cases of desktop vs. mobile rendering of the floating container. Oh and also putting a few of the different types of things in it that it supports.

Reply:
I am thinking this set?
```
{category: 'normal-screen-size', tests: ['isFloating']},
{category: 'mobile-screen-size', tests: ['isFloating']},
{category: 'normal-screen-size-rtl', tests: ['isFloating']},
{category: 'mobile-screen-size-rtl', tests: ['isFloating']}
```

Reply:
Not sure RTL needs to be tested with both desktop and mobile.  I think the interesting thing about mobile is when/whether floating will occur.